### PR TITLE
add article and solver for day 1 2022

### PR DESCRIPTION
**Pull request by bishabosha**
proof of concept

### CODE REVIEW

File: docs/2022/puzzles/day01.md [hunk 1]
```diff
@@ -1,11 +1,75 @@
...
 https://adventofcode.com/2022/day/1
 
+## Solution Summary
+
+First transform the input into a `List` of `Inventory`, each `Inventory` is a list of `Int`, representing the calorie
+count of an item in the inventory, this is handled in `scanInventories`.
+
+### Part 1
+
+given the `List` of `Inventory`, we must first find the total calorie count of each inventory.
```
Comment (adpi2) [inline docs/2022/puzzles/day01.md:17]:
```suggestion
Given the `List` of `Inventory`, we must first find the total calorie count of each inventory.
```

File: docs/2022/puzzles/day01.md [hunk 2]
```diff
@@ -1,11 +1,75 @@
...
+
+> For a single `Inventory`, we do this with the `sum` method its `items` property (found in the `List` class). e.g. `inventory.items.sum`.
```
Comment (adpi2) [inline docs/2022/puzzles/day01.md:19]: Why is this in a quote `>` ?

Also I think there is something missing in the sentence:
```suggestion
For a single `Inventory`, we do this using the `sum` method on its `items` property (found in the `List` class). e.g. `inventory.items.sum`.
```

File: docs/2022/puzzles/day01.md (hunk 1 above)
Comment (bishabosha) [reply docs/2022/puzzles/day01.md]: thanks

File: docs/2022/puzzles/day01.md (hunk 2 above)
Comment (bishabosha) [reply docs/2022/puzzles/day01.md]: thanks

**Pull request merged**